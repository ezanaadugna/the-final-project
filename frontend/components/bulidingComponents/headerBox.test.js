import React from "react";
import { render } from "@testing-library/react-native";
import HeaderBoxComponent from "./headerBox";

describe("HeaderBoxComponent", () => {
  // Test to check if the component renders correctly
  it("renders correctly", () => {
    const { toJSON } = render(<HeaderBoxComponent text="Test Header" />);
    expect(toJSON()).toMatchSnapshot();
  });

  // Test to check if the correct text is displayed
  it("displays the correct text", () => {
    const { getByText } = render(<HeaderBoxComponent text="Test Header" />);
    const headerText = getByText("Test Header");
    expect(headerText).toBeDefined();
  });
});
