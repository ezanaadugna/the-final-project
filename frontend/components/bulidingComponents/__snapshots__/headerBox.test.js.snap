// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBoxComponent renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 10,
      "elevation": 3,
      "height": "10%",
      "justifyContent": "center",
      "marginTop": 40,
      "padding": 10,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 4,
      "width": "90%",
    }
  }
>
  <Text
    style={
      {
        "color": "black",
        "fontFamily": "Helvetica",
        "fontSize": 20,
      }
    }
  >
    Test Header
  </Text>
</View>
`;
